### PR TITLE
fix: return type is not mentioned correctly in useOne hook

### LINKED ISSUE #5312: [FEAT] Missing HttpError error type in useOne hooks.
### Is your feature request related to a problem? Please describe.

I noticed that the TError generic is not passed into the QueryObserverResult in the useOne hook, which results in a type inference of unkown in useOne . Eventually the type is lost.I was wondering if this is by design? Or if there is a plan to fix it?

https://github.com/refinedev/refine/blob/master/packages/core/src/hooks/data/useOne.ts#L109

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Describe the thing to improve

Useone hook's return type is incorrect.

### PR DESCRIPTION
**Pull request by ksankeerth**
This PR fixes #5312 . Return type is not mentioned properly in useOne hook. 

Explain the **details** for making this change. What existing problem does the pull request solve?

### Test plan (required)

Tested tests written for useOne hook and didn't fail

### Closing issues
closes #5312 

### Self Check before Merge

Please check all items below before review.

-   [X] Corresponding issues are created/updated or not needed
-   [X] Docs are updated/provided or not needed
-   [X] Examples are updated/provided or not needed
-   [X] TypeScript definitions are updated/provided or not needed
-   [X] Tests are updated/provided or not needed
-   [X] Changesets are provided or not needed

### CODE REVIEW

**Review by BatuhanW — APPROVED** [review]
LGTM 🚀

**Review by alicanerdurmaz — APPROVED** [review]

**Comment by BatuhanW** [comment]
Hello @ksankeerth thanks for the contribution! We've noticed we are missing TError on more hooks such as `useMany`, `useForm`, `useShow`, and `useSelect`. Would you like to include them in your PR? If not, we'll create an issue for the community.

**Comment by ksankeerth** [comment]
> Hello @ksankeerth thanks for the contribution! We've noticed we are missing TError on more hooks such as `useMany`, `useForm`, `useShow`, and `useSelect`. Would you like to include them in your PR? If not, we'll create an issue for the community.

Sure. I'll update my PR with fixes for other hooks.

**Comment by ksankeerth** [comment]
> Hello @ksankeerth thanks for the contribution! We've noticed we are missing TError on more hooks such as `useMany`, `useForm`, `useShow`, and `useSelect`. Would you like to include them in your PR? If not, we'll create an issue for the community.

done

File: documentation/docs/api-reference/core/hooks/data/useMany/index.md [hunk 1]
```diff
@@ -294,10 +294,10 @@ errorNotification-default='"Error (status code: `statusCode`)"'
 
 ### Return values
 
-| Description                               | Type                                                                                                     |
-| ----------------------------------------- | -------------------------------------------------------------------------------------------------------- |
-| Result of the TanStack Query's `useQuery` | [`QueryObserverResult<{ data: TData[]; }>`](https://tanstack.com/query/v4/docs/react/reference/useQuery) |
-| overtime                                  | `{ elapsedTime?: number }`                                                                               |
+| Description                               | Type                                                                                                                  |
+| ----------------------------------------- | --------------------------------------------------------------------------------------------------------------------- |
+| Result of the TanStack Query's `useQuery` | [`QueryObserverResult<{ data: TData[];error: TError }>`](https://tanstack.com/query/v4/docs/react/reference/useQuery) |
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/data/useMany/index.md:299]:
```suggestion
| Result of the TanStack Query's `useQuery` | [`QueryObserverResult<{ data: TData[]; error: TError }>`](https://tanstack.com/query/v4/docs/react/reference/useQuery) |
```

**Review by omeraplak — APPROVED** [review]

**Pull request merged**